client/src/components/Dashboard.js

```diff
@@ -1,36 +1,94 @@
-import React, { Component } from 'react';
-import { Button, Row, Col, Icon, Input, Navbar, NavItem, Dropdown } from 'react-materialize';
+import React, { Component, PropTypes } from 'react';
+import { Row, Pagination } from 'react-materialize';
+import { browserHistory } from 'react-router';
+import { connect } from 'react-redux';
+import NavBar from './navbar/navBar';
+import DocumentForm from './documents/documentForm';
+import DocumentPreview from './documents/documentPreview';
+import Filter from './filter/documentFilter';
 import '../main.scss';
+import { isLoggedIn } from '../helpers/auth';
+import { fetchUserDocument, pagination } from '../actions/actionCreator';
 
-class Dashboard extends React.Component {
+class Dashboard extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      metadata: [],
+      result: []
+    };
+    this.displayData = this.displayData.bind(this);
+  }
+
+  componentDidMount() {
+    if (!isLoggedIn()) {
+      browserHistory.push('/');
+    } else {
+      this.props.fetchUserDocument();
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.metadata.hasOwnProperty('metadata')) {
+      const { metadata, result } = nextProps.metadata;
+      this.setState({ metadata, result });
```

Comment:
I think you can proceed to use these values from the `props` directly instead of setting the state. States generally have more overhead than props.